### ISSUE TITLE
Remove segments while editing a track

### ISSUE THREAD

**Issue by vsassone** [osmandapp/OsmAnd#17516]
While editing a map, it would be a useful addition to be able to delete all points between two given ones, as one often wants to clean a recorded track up by removing detours, misturns, etc. Haven’t found a way to do that apart from splitting the track in three, deleting the unwanted segment, then “join” back the remaining two. This is laborious, error prone, and the result is not really the same. 

Cheers and thanks.

**Comment by vshcherb** [comment]
CC @DmitryAlexei  I think It's already possible  to do it, will get back with more inof

**Comment by vshcherb** [comment]
Suggestion: scroll in list while selecting point on the map

**Comment by vsassone** [comment]
Thanks, what list? The list of points? It’s very unwieldy, I couldn’t even select a specific point on it without scrolling over thousands of them...

On 30 Jun 2023, at 14:29, vshcherb ***@***.***> wrote:

CAUTION: This e-mail originated outside the University of Southampton.


Suggestion: scroll in list while selecting point on the map

**Comment by DmitryAlexei** [comment]
At the moment, you can delete a fragment from a track using the method described above or by deleting points. To make working with the track easier, the number of points can be reduced using the GPS filter.
(see https://osmand.net/docs/user/map/track-context-menu#gps-filter).

However, as correctly noted, to find the right point you sometimes have to scroll through a long list of them. And when you tap on the point either on the map or in the list, *[Point context menu](https://osmand.net/docs/user/plan-route/create-route#point-context-menu)* pops up, and you cannot see the whole picture. 

Suggestions:
- Perhaps we can add the ability to **select a specific point from the list on the track, as in the graph tab**? (when you can tap on a point from the list and see its place on the track and vice versa by clicking on a point on the track to see its place in the list. Point context menu pops up only when you tap on the specific place in the list) 
- In addition to the existing "add new segment" function, **implement the "allocate segment"** function in the options? (you need to take two points from the track to create a segment) 

<table style="width:100%">
  <tr>
<th><video src="https://github.com/osmandapp/OsmAnd/assets/116079541/9583faa5-4ec5-4aab-b3f6-95f959c03d4f"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/77f28317-3014-4df4-b2c5-7a7dcf20d87e"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by vsassone** [comment]
Apologies, I’m not sure Im getting the suggestions here. The way to delete a portion of a track is to (1) split it at the two points, (2) exit the edit function and delete the mid section, (3) use the join segments option. This works but isn’t perfect because it leaves a track in two segments, and if you repeat it multiple times it leaves you with a track highly segmented where the segments have no logical sense. 

Sound like it’s been suggested above that this can be done through the point list. Interesting. Firstly, I don’t know how to efficiently select a point of the list, say by clicking on the track on the map. I have to (1) click on the map, (2) see the number associated with the first point to be deleted, (3) click again to (4) see the number of the last point to be deleted, (4) go to the list of points, (5) scroll down with a finger (say I’m on a phone or tablet) potentially thousands and thousands of points to reach the first point, (6) repeatedly click on the X on the side until I reach the final point, which again could be hundreds and hundreds of times. Is this what is been suggested? Very laborious and unwieldy, hope I’m missing a trick here because this not viable. Much better to split the track into segments, tbh. 

(Aside comment: would be nice to have am option in the context menu to move from a point on the map to the same point in the list or graph.)

Perhaps a simple way to implement the functionality i’m after would be to add a way to “merge segments”, not simply “join” them, so that the results of a merge is to reduce two segments into a single one. This could be used after the “splitting and segment deleting technique” above to bring back the track to the correct number of segments as desired by the designer. 

 Cheers.

**Comment by DmitryAlexei** [comment]
I agree with the above. A little remark about working with points of a track before deleting them manually.

You can reduce the number of points of the track using **GPS filter** **beforehand** (see https://osmand.net/docs/user/map/track-context-menu#gps-filter).
**GPS filter** can help you get a more accurate graph and visual route line without distortion or recording noise.

**Comment by vsassone** [comment]
Not sure about how the gps filter works, will check it out. Does it mask points temporarily, or rather removes them for good?

**Comment by scaidermern** [comment]
It removes points permanently if you save the track.

**Comment by jpravetz** [comment]
> Apologies, I’m not sure Im getting the suggestions here. The way to delete a portion of a track is to (1) split it at the two points, (2) exit the edit function and delete the mid section, (3) use the join segments option. This works but isn’t perfect because it leaves a track in two segments, and if you repeat it multiple times it leaves you with a track highly segmented where the segments have no logical sense.
> 
> Sound like it’s been suggested above that this can be done through the point list. Interesting. Firstly, I don’t know how to efficiently select a point of the list, say by clicking on the track on the map. I have to (1) click on the map, (2) see the number associated with the first point to be deleted, (3) click again to (4) see the number of the last point to be deleted, (4) go to the list of points, (5) scroll down with a finger (say I’m on a phone or tablet) potentially thousands and thousands of points to reach the first point, (6) repeatedly click on the X on the side until I reach the final point, which again could be hundreds and hundreds of times. Is this what is been suggested? Very laborious and unwieldy, hope I’m missing a trick here because this not viable. Much better to split the track into segments, tbh.
> 
> (Aside comment: would be nice to have am option in the context menu to move from a point on the map to the same point in the list or graph.)
> 
> Perhaps a simple way to implement the functionality i’m after would be to add a way to “merge segments”, not simply “join” them, so that the results of a merge is to reduce two segments into a single one. This could be used after the “splitting and segment deleting technique” above to bring back the track to the correct number of segments as desired by the designer.
> 
> Cheers.

This describes exactly how cumbersome it is to delete a mid section. Splitting and trimming is a poor workaround. Instead just improve your app please. Specifically:
- either allow a selection of points, on the map, for group deletion
- or make clicking on a point highlight that point in the point list
- and then make it easier to select the first and last point, allowing a person to delete every point in between.

In addition I find it confusing and disorienting that, while wanting to delete points,  the UI is showing a point that can be added to the end of the track.  I hopme you can come to the conclusion that this is a bad idea.
